lib/assert.js

```diff
@@ -416,7 +416,9 @@ function _throws(shouldThrow, block, expected, message) {
       userProvidedMessage &&
       expectedException(actual, expected)) ||
       isUnexpectedException) {
-    fail(actual, expected, 'Got unwanted exception' + message);
+    fail(actual, expected,
+         'Got unwanted exception:\n' +
+         actual.message + message);
```

Comment:
Looks really good. 
But probably something like that 

```
'Got unwanted exception' + message + '\n' + actual.message
```
(without **:** after `exception`) since `message` will have it (if message exists). Otherwise message will have just **.** (dot). 

So, if `message` wasn't passed it will look like:

```
Got unwanted exception.
Message of the unwanted exception.
```

I really like your idea.